README.md

```diff
@@ -11,13 +11,16 @@ Multilingual **Uni**t **T**est and Function Source **Sync**hronization for CodeL
 
 To run this script on a new project, you need to install the corresponding language server:
 
-| Language | Language Server                                              | Supported |
-| -------- | ------------------------------------------------------------ | --------- |
-| Python   | [pylsp](https://github.com/python-lsp/python-lsp-server)     | &#x2714;  |
-| C/C++    | [clangd](https://clangd.llvm.org/installation.html)          | &#x2714;  |
-| Rust     | [rust-analyzer](https://rust-analyzer.github.io/manual.html) | ToDo      |
+| Language | Language Server                                              | Frontend | Backend  |
+| -------- | ------------------------------------------------------------ | -------- | -------- |
+| Python   | [pylsp](https://github.com/python-lsp/python-lsp-server)     | &#x2714; | &#x2714; |
+| C/C++    | [clangd](https://clangd.llvm.org/installation.html)          | ToDo     | &#x2714; |
+| Rust     | [rust-analyzer](https://rust-analyzer.github.io/manual.html) | ToDo     | ToDo     |
 
 Other languages are not supported yet, but will be as the research progresses.
+To support a new langauge, you need a frontend to do the following:
+1. Collect the unit tests locations and focal functions locations in the repo (see `scripts/collect_test.py` and `scripts/collect_focal.py` for Python frontend).
```

Comment:
@DataCorrupted  Since [CodeSearchNet](https://arxiv.org/abs/1909.09436) has 6 languages Go, Java, JavaScript, Python, PHP, and Ruby. I will follow their paper to use [tree-sitter](https://tree-sitter.github.io/tree-sitter/) to re-implement/refactor these two steps. After that, this project will support Rust automatically since it is supported by tree-sitter.